rfd/0153-resource-guidelines.md

```diff
@@ -0,0 +1,742 @@
+---
+authors: Tim Ross (tim.ross@goteleport.com)
+state: draft
+---
+
+# RFD 153 - Resource Guidelines
+
+## Required approvers
+
+- Engineering: `@zmb3`
+
+## What
+
+Guidelines for creating backend resources and APIs to interact with them.
+
+## Why
+
+To date resources follow slightly different patterns and have inconsistent APIs for creating, updating and deleting
+them, see [#29234](https://github.com/gravitational/teleport/issues/29234). In addition to reducing confusion and
+fostering a better developer experience, this will also remove complexity from the terraform provider and teleport
+operator that have to accommodate for every subtle API and resource difference that exist today.
+
+## Details
+
+### Project structure
+
+All protos are defined in the api module under the proto directory. When adding a new resource and gRPC API a new folder
+that matches the desired package name of the proto, generally a friendly name of the resource, should be created. Inside
```

Comment:
Nit: In my head, I quite like the word "domain" to describe the appropriate groupings - we probably don't necessarily expect a package per resource where several resources are tightly coupled/related.